backend/app/routers/experiments.py

```diff
@@ -141,37 +149,72 @@ async def get_experiment_runs(
 
 
 @router.get("/count/experiments/{id}/runs", response_model=int)
-async def get_experiment_runs_count(id: PydanticObjectId) -> Any:
+async def get_experiment_runs_count(
+    id: PydanticObjectId,
+    user: dict = Depends(get_current_user(required=False)),
+) -> Any:
+    await check_experiment_access(id, user)
+
     return await ExperimentRun.find(ExperimentRun.experiment_id == id).count()
 
 
 @router.get("/experiment-runs", response_model=list[ExperimentRunResponse])
-async def get_all_experiment_runs(pagination: Pagination = Depends()) -> Any:
-    runs = await ExperimentRun.find_all(
-        skip=pagination.offset, limit=pagination.limit
-    ).to_list()
+async def get_all_experiment_runs(
+    pagination: Pagination = Depends(),
+    user: dict = Depends(get_current_user(required=False)),
```

Comment:
This is also required=True

Reply:
Are you sure you want all of those? The idea was that this is ready for public experiments. We are checking the access by `check_experiment_access()` anyway.

Reply:
I'm wondering if it's not better to lock it now (when there are no public experiments) and then unlock it once public experiments are introduced?